### PR TITLE
AGR-1925 new filte

### PR DESCRIPTION
**Pull request by FB-ping**
here to add more filter to genetic interaction  and also fix the bug to display phenotype

### CODE REVIEW

File: agr_api/.temp-Main (1)-classpath-arg-1601052883005.txt [hunk 1]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1601052883005.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1602613710831.txt [hunk 2]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1602613710831.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1602613961020.txt [hunk 3]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1602613961020.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1602621065242.txt [hunk 4]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1602621065242.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1609302021441.txt [hunk 5]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1609302021441.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1609774301952.txt [hunk 6]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1609774301952.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1609774427754.txt [hunk 7]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1609774427754.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1609774670769.txt [hunk 8]
```diff
@@ -0,0 +1 @@
...ers\pingl\.m2\repository\io\netty\netty-handler\4.1.45.Final\netty-handler-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-common\4.1.45.Final\netty-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-buffer\4.1.45.Final\netty-buffer-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport\4.1.45.Final\netty-transport-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-resolver\4.1.45.Final\netty-resolver-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-codec\4.1.45.Final\netty-codec-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-epoll\4.1.45.Final\netty-transport-native-epoll-4.1.45.Final-linux-x86_64.jar;C:\Users\pingl\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.45.Final\netty-transport-native-unix-common-4.1.45.Final.jar;C:\Users\pingl\.m2\repository\com\google\code\gson\gson\2.8.5\gson-2.8.5.jar;C:\Users\pingl\.m2\repository\org\infinispan\infinispan-commons\10.1.3.Final\infinispan-commons-10.1.3.Final.jar;C:\Users\pingl\.m2\repository\com\opencsv\opencsv\5.1\opencsv-5.1.jar;C:\Users\pingl\.m2\repository\org\apache\commons\commons-text\1.8\commons-text-1.8.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1609774670769.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1610423744091.txt [hunk 9]
```diff
@@ -0,0 +1 @@
...m2\repository\com\google\inject\guice\4.1.0\guice-4.1.0-no_aop.jar;C:\Users\pingl\.m2\repository\aopalliance\aopalliance\1.0\aopalliance-1.0.jar;C:\Users\pingl\.m2\repository\org\codehaus\groovy\groovy-xml\3.0.5\groovy-xml-3.0.5.jar;C:\Users\pingl\.m2\repository\junit\junit\4.13.1\junit-4.13.1.jar;C:\Users\pingl\.m2\repository\org\hamcrest\hamcrest-core\1.3\hamcrest-core-1.3.jar;C:\Users\pingl\.m2\repository\org\apache\logging\log4j\log4j-api\2.13.2\log4j-api-2.13.2.jar;C:\Users\pingl\.m2\repository\org\spockframework\spock-core\2.0-M3-groovy-3.0\spock-core-2.0-M3-groovy-3.0.jar;C:\Users\pingl\.m2\repository\org\junit\platform\junit-platform-engine\1.7.0-M1\junit-platform-engine-1.7.0-M1.jar;C:\Users\pingl\.m2\repository\org\opentest4j\opentest4j\1.2.0\opentest4j-1.2.0.jar;C:\Users\pingl\.m2\repository\org\junit\platform\junit-platform-commons\1.7.0-M1\junit-platform-commons-1.7.0-M1.jar;C:\Users\pingl\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.6.2\junit-jupiter-engine-5.6.2.jar;C:\Users\pingl\.m2\repository\org\apiguardian\apiguardian-api\1.1.0\apiguardian-api-1.1.0.jar;C:\Users\pingl\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.6.2\junit-jupiter-api-5.6.2.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1610423744091.txt:1]: This files need to be not added to the code base.

File: agr_api/.temp-Main (1)-classpath-arg-1610423984088.txt [hunk 10]
```diff
@@ -0,0 +1 @@
...m2\repository\com\google\inject\guice\4.1.0\guice-4.1.0-no_aop.jar;C:\Users\pingl\.m2\repository\aopalliance\aopalliance\1.0\aopalliance-1.0.jar;C:\Users\pingl\.m2\repository\org\codehaus\groovy\groovy-xml\3.0.5\groovy-xml-3.0.5.jar;C:\Users\pingl\.m2\repository\junit\junit\4.13.1\junit-4.13.1.jar;C:\Users\pingl\.m2\repository\org\hamcrest\hamcrest-core\1.3\hamcrest-core-1.3.jar;C:\Users\pingl\.m2\repository\org\apache\logging\log4j\log4j-api\2.13.2\log4j-api-2.13.2.jar;C:\Users\pingl\.m2\repository\org\spockframework\spock-core\2.0-M3-groovy-3.0\spock-core-2.0-M3-groovy-3.0.jar;C:\Users\pingl\.m2\repository\org\junit\platform\junit-platform-engine\1.7.0-M1\junit-platform-engine-1.7.0-M1.jar;C:\Users\pingl\.m2\repository\org\opentest4j\opentest4j\1.2.0\opentest4j-1.2.0.jar;C:\Users\pingl\.m2\repository\org\junit\platform\junit-platform-commons\1.7.0-M1\junit-platform-commons-1.7.0-M1.jar;C:\Users\pingl\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.6.2\junit-jupiter-engine-5.6.2.jar;C:\Users\pingl\.m2\repository\org\apiguardian\apiguardian-api\1.1.0\apiguardian-api-1.1.0.jar;C:\Users\pingl\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.6.2\junit-jupiter-api-5.6.2.jar
```
Comment (oblodgett) [inline agr_api/.temp-Main (1)-classpath-arg-1610423984088.txt:1]: This files need to be not added to the code base.

File: agr_api/src/main/java/org/alliancegenome/api/service/GeneService.java [hunk 11]
```diff
@@ -31,6 +32,8 @@
 
     @Inject
     private AlleleService alleleService;
+    @Inject
+    private CacheService cacheService;
```
Comment (oblodgett) [inline agr_api/src/main/java/org/alliancegenome/api/service/GeneService.java:36]: This `cacheService` should not be called directly. It should be called through the GeneCacheRepository in org.alliancegenome.cache.repository.

Comment (oblodgett) [reply agr_api/src/main/java/org/alliancegenome/api/service/GeneService.java:36]: Create a method in there GeneCacheRepository.getInteractions(id);

**Review by oblodgett — CHANGES_REQUESTED** [review]

**Comment by FB-ping** [comment]
fixed, created new method in GeneCacheRepository List<InteractionGeneJoin> getInteractions(String geneID)

File: agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/InteractionRepository.java [hunk 12]
```diff
@@ -47,11 +68,20 @@ public long getInteractorCount(String geneID) {
...
+        String allInteractionsQuery = "MATCH p1=(g1:Gene)--(igj:InteractionGeneJoin)--(g2:Gene), p2=(igj:InteractionGeneJoin)--(s) "
+                + " OPTIONAL MATCH p3=(g1:Gene)-->(s1:Species) "
+                + " OPTIONAL MATCH p4=(g2:Gene)-->(s2:Species) ";
         //allInteractionsQuery += " where g1.primaryKey = 'MGI:109583' ";
         //allInteractionsQuery += " where g1.primaryKey = 'MGI:103170' ";
         //allInteractionsQuery += " where g1.primaryKey = 'FB:FBgn0029891' ";
-        String query = allInteractionsQuery + " RETURN p ";
+        //String query = allInteractionsQuery + " RETURN p ";
+        //String query = allInteractionsQuery + " RETURN p1, p2, p3, p4";
+        //String query = "MATCH p1=(g1:Gene)--(igj:InteractionGeneJoin)--(g2:Gene), p2=(igj:InteractionGeneJoin)--(s) where g1.primaryKey = 'WB:WBGene00000390'"
```
Comment (oblodgett) [inline agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/InteractionRepository.java:80]: May want to clean up all these comments and duplicated query?

**Review by oblodgett — APPROVED** [review]

**Pull request merged**